src/components/TradeWidget/index.tsx

```diff
@@ -64,17 +81,37 @@ const TradeWidget: React.FC = () => {
   const tokens = useMemo(() => tokenListApi.getTokens(fallBackNetworkId), [fallBackNetworkId])
 
   // Listen on manual changes to URL search query
-  const { sell: sellTokenSymbol, receive: receiveTokenSymbol } = useParams()
+  const { sell: sellTokenSymbol, buy: receiveTokenSymbol } = useParams()
+  const query = useQuery()
+  const sellAmount = query.get('sell')
+  const receiveAmount = query.get('buy')
 
   const [sellToken, setSellToken] = useState(
     () => getToken('symbol', sellTokenSymbol, tokens) || getToken('symbol', 'DAI', tokens),
   )
   const [receiveToken, setReceiveToken] = useState(
     () => getToken('symbol', receiveTokenSymbol, tokens) || getToken('symbol', 'USDC', tokens),
   )
+  const sellInputId = 'sellToken'
+  const receiveInputId = 'receiveToken'
 
-  // Change URL on internal token change
-  useURLParams(`sell=${sellToken.symbol}-0&receive=${receiveToken.symbol}-20`)
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const sanitizedSellAmount = useMemo(() => sanitizeInputAmount(sellAmount, sellToken.decimals), [])
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const sanitizedReceiveAmount = useMemo(() => sanitizeInputAmount(receiveAmount, receiveToken.decimals), [])
+
+  const methods = useForm({
+    mode: 'onBlur',
+    defaultValues: {
+      [sellInputId]: sanitizedSellAmount,
+      [receiveInputId]: sanitizedReceiveAmount,
+    },
+  })
+  const { handleSubmit, watch, reset } = methods
+
+  const urlParams = new URLSearchParams({ sell: watch(sellInputId), buy: watch(receiveInputId) })
+  const url = `/trade/${sellToken.symbol}-${receiveToken.symbol}?${urlParams.toString()}`
```

Comment:
nit: I might be wrong, but I would think that if you drop the `toString()`, it has the same effect

Reply:
Yeah it works! Refactored.